src/Symfony/Component/Messenger/Stamp/ErrorDetailsStamp.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Stamp;
+
+use Symfony\Component\ErrorHandler\Exception\FlattenException;
+use Symfony\Component\Messenger\Exception\HandlerFailedException;
+use Throwable;
+
+/**
+ * Stamp applied when a messages fails due to an exception in the handler.
+ */
+final class ErrorDetailsStamp implements StampInterface
+{
+    /** @var string */
+    private $exceptionClass;
+
+    /** @var int|mixed */
+    private $exceptionCode;
+
+    /** @var string */
+    private $exceptionMessage;
+
+    /** @var FlattenException|null */
+    private $flattenException;
+
+    public function __construct(Throwable $throwable)
+    {
+        if ($throwable instanceof HandlerFailedException) {
+            $throwable = $throwable->getPrevious();
```

Comment:
Nicer way to get the first nested exception that before 👍

Reply:
Thanks.